src/Domain/Domain.cpp

```diff
@@ -45,6 +49,41 @@ Domain<VolumeDim, TargetFrame>::Domain(
   }
 }
 
+template <size_t VolumeDim, typename TargetFrame>
+Domain<VolumeDim, TargetFrame>::Domain(
```

Comment:
I very much have the same issue as @markscheel. Why can't you just pass all the blocks to this?